### PR TITLE
lk2nd: display: panel: Introduce custom oem_panel implementation

### PR DESCRIPTION
**Pull request by wonderfulShrineMaidenOfParadise**

### CODE REVIEW

File: lk2nd/display/panel/generated/lk_panel_hx8394d_720p_video.h [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+};
+
+static char hx8394d_720p_video_off_cmd_0[] = {
+	0x28, 0x00, 0x05, 0x80
+};
+static char hx8394d_720p_video_off_cmd_1[] = {
+	0x10, 0x00, 0x05, 0x80
+};
+
+static struct mipi_dsi_cmd hx8394d_720p_video_off_command[] = {
```
Comment (wonderfulShrineMaidenOfParadise) [inline lk2nd/display/panel/generated/lk_panel_hx8394d_720p_video.h:186]:
```
lk2nd/display/panel/generated/lk_panel_hx8394d_720p_video.h:186:28: warning: 'hx8394d_720p_video_off_command' defined but not used [-Wunused-variable]
  186 | static struct mipi_dsi_cmd hx8394d_720p_video_off_command[] = {
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: lk2nd/display/panel/generated/lk_panel_ili9806e_boyi_byt45mp41t1m_4p5cpt.h [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
+};
+
+static char ili9806e_boyi_byt45mp41t1m_4p5cpt_off_cmd_0[] = {
+	0x28, 0x00, 0x05, 0x80
+};
+static char ili9806e_boyi_byt45mp41t1m_4p5cpt_off_cmd_1[] = {
+	0x10, 0x00, 0x05, 0x80
+};
+
+static struct mipi_dsi_cmd ili9806e_boyi_byt45mp41t1m_4p5cpt_off_command[] = {
```
Comment (wonderfulShrineMaidenOfParadise) [inline lk2nd/display/panel/generated/lk_panel_ili9806e_boyi_byt45mp41t1m_4p5cpt.h:541]:
```
lk2nd/display/panel/generated/lk_panel_ili9806e_boyi_byt45mp41t1m_4p5cpt.h:541:28: warning: 'ili9806e_boyi_byt45mp41t1m_4p5cpt_off_command' defined but not used [-Wunused-variable]
  541 | static struct mipi_dsi_cmd ili9806e_boyi_byt45mp41t1m_4p5cpt_off_command[] = {
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: lk2nd/display/panel/oem_panel.c [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+uint32_t oem_panel_max_auto_detect_panels()
+{
+	return 0;
+}
+
+#define _panel_select(panel)	panel_##panel##_select
+#define panel_select(panel)	_panel_select(panel)
+
+int oem_panel_select(const char *panel_name, struct panel_struct *panel,
```
Comment (wonderfulShrineMaidenOfParadise) [inline lk2nd/display/panel/oem_panel.c:45]:
```
lk2nd/display/panel/oem_panel.c:45:5: warning: no previous prototype for 'oem_panel_select' [-Wmissing-prototypes]
   45 | int oem_panel_select(const char *panel_name, struct panel_struct *panel,
      |     ^~~~~~~~~~~~~~~~
```

**Review by wonderfulShrineMaidenOfParadise — COMMENTED** [review]
Some warnings.

File: lk2nd/display/panel/generated/lk_panel_hx8394d_720p_video.h [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+	panel->panelres = &hx8394d_720p_video_panel_res;
+	panel->color = &hx8394d_720p_video_color;
+	panel->videopanel = &hx8394d_720p_video_video_panel;
+	panel->commandpanel = &hx8394d_720p_video_command_panel;
+	panel->state = &hx8394d_720p_video_state;
+	panel->laneconfig = &hx8394d_720p_video_lane_config;
+	panel->paneltiminginfo = &hx8394d_720p_video_timing_info;
+	panel->panelresetseq = &hx8394d_720p_video_reset_seq;
+	panel->backlightinfo = &hx8394d_720p_video_backlight;
+	pinfo->mipi.panel_on_cmds = hx8394d_720p_video_on_command;
```
Comment (TravMurav) [inline lk2nd/display/panel/generated/lk_panel_hx8394d_720p_video.h:257]:
```suggestion
	pinfo->mipi.panel_on_cmds = hx8394d_720p_video_on_command;
	pinfo->mipi.panel_off_cmds = hx8394d_720p_video_off_command;
```

Can you check if this doesn't break the display? Not sure why lmdpdg doesn't generate this...

Comment (wonderfulShrineMaidenOfParadise) [reply lk2nd/display/panel/generated/lk_panel_hx8394d_720p_video.h:257]: Link: msm8916-mainline/linux-mdss-dsi-panel-driver-generator#48

**Comment by TravMurav** [comment]
Thanks!

**Pull request merged**